lib/Parse/ParseExpr.cpp

```diff
@@ -2778,7 +2778,14 @@ ExprResult Parser::ParseInteropTypeAnnotation() {
     if (PT.expectAndConsume(diag::err_expected_lparen_after,
                             Ident->getNameStart()))
       return ExprError();
-    TypeResult Ty = ParseTypeName();
+    // Interop type annotations for parameter declarators must be parsed in a
+    // special context.  Array types may have static or a type qualifier
+    // declared as part of the first dimension, that is,
+    //     int a[static 10] : itype(int [static 10])
```

Comment:
I'm probably missing the connection here; is it that `static` or other type qualifiers are only present in the `Declarator`'s context because they're stripped out otherwise?